src/ftw_cli/download_img.py

```
@@ -8,10 +8,12 @@
 import xarray as xr
 from tqdm.auto import tqdm
 from shapely.geometry import shape
+from tenacity import retry, wait_random_exponential, stop_after_attempt
 
 from .cfg import BANDS_OF_INTEREST, COLLECTION_ID, MSPC_URL
 
 
+@retry(wait=wait_random_exponential(max=30), stop=stop_after_attempt(5))
```

What kind of error do we get from MS PC?

For the server, a smaller wait time would be better (like 3-5 secs or so). Is there any way we could configgure this?

---

Yes tenacity is retrying this function call up to 5 times, waiting roughtly `2^n` seconds between attempts where `n` is the number of attempts.  It's not exactly exponential, it injects some randomness (jitter) into the waits.  This sort of "exponential backoff with jitter" is pretty standard practice in distributed systems.

> I assume the wait only applies for non-successful calls?

Correct.  Tenacity only retries failed calls (unhandled exceptions).

> if we would set the timeout to 5 secs and you'd receive a 200 within a second and download takes 10 seconds, that would not abort download due to the wait timeout, right?

We could set a timeout, although that would be handled through the underlying IO library used by `pystac.Item.from_file`, not tenacity.  Tenacity itself doesn't implement timeouts, it just provides retry/backoff logic.

---

This would let us "fail fast", but still recover from some amount of PC inavailability.

---

Yeah, sounds good to me!